src/TSTransformer/util/offset.ts

```diff
@@ -16,6 +18,26 @@ export function offset(expression: luau.Expression, value: number) {
 	if (value === 0) {
 		return expression;
 	}
+	// this special case handles when the offset function is called next to a binary expression that has a number literal on its right
+	// IE: array[offset - 1 + 1] -> array[offset]
+	if (
+		luau.isBinaryExpression(expression) &&
+		luau.isNumberLiteral(expression.right) &&
+		(expression.operator === "+" || expression.operator === "-")
+	) {
+		const rightValue = getLiteralNumberValue(expression.right);
```

Comment:
I wonder if this might be more streamlined by sequentially counting the various values in the highest-up binaryExp chain, and then rewriting it to its simplest form, similar to how the compiler does with optional chains or some logicals.